packages/vm/src/context.rs

```diff
@@ -399,6 +415,17 @@ mod test {
         move_into_context(ctx, storage, querier);
     }
 
+    /// This is a testing-only implementation that panics on overconsumption.
+    /// We currently don't have a production-ready version of this since only Wasmer consumes VM gas
+    /// directly. This might change in the future (https://github.com/CosmWasm/cosmwasm/pull/475)
+    fn decrease_gas_left(instance: &mut WasmerInstance, amount: u64) {
+        let current_limit = get_gas_left(instance);
+        let new_limit = current_limit
+            .checked_sub(amount)
+            .expect("Must not decrease more than available");
+        set_gas_limit(instance, new_limit);
```

Comment:
I thought this was not doing  `metering::set_points_used`  but then read the code and this is exactly what you want here.

Sometimes confusing when I have two levels of abstraction (wasmer and cosmwasm_vm::backend) in my head at once with different names. But this is good as it is.